src/Symfony/Component/Validator/Constraints/EmailValidator.php

```diff
@@ -37,7 +37,7 @@ public function validate($value, Constraint $constraint)
         $valid = filter_var($value, FILTER_VALIDATE_EMAIL);
 
         if ($valid) {
-            $host = substr($value, strpos($value, '@') + 1);
+            $host = substr(strrchr($value, '@'), 1);
```

Comment:
`substr($value, strrpos($value, '@') + 1);` would be faster (less string copies)